### PR TITLE
Replace jquery.expander using jquery.toggle

### PR DESCRIPTION
**Pull request by mdeniz**
Created a helper to render both texts, the real one and a sliced version of it and manage the show/hide operations using jQuery.toggle

### CODE REVIEW

**Comment by Ana06** [comment]
[(3) Drop jquery.expander](https://trello.com/c/BI9qVAMc/208-3-drop-jquery-expander)

File: src/api/app/helpers/webui/webui_helper.rb [hunk 1]
```diff
@@ -411,4 +411,18 @@ def word_break(string, length)
     # adds a <wbr> tag after an amount of given characters
     safe_join(string.scan(/.{1,#{length}}/), "<wbr>".html_safe)
   end
+
+  def toggle_sliced_text(text, slice_length = 50, id = nil)
+    return text if text.length < slice_length
+    id ||= "toggle_sliced_text_#{rand(9**10)}"
```
Comment (Ana06) [inline src/api/app/helpers/webui/webui_helper.rb]: Do we really need a random id of 10 digits?

Comment (Ana06) [inline src/api/app/helpers/webui/webui_helper.rb]: Do we really need a random number of 10 digits for the `id`?

Comment (evanrolfe) [inline src/api/app/helpers/webui/webui_helper.rb]: maybe use a timestamp instead? thats how railscasts do it, see `add_fields()` in application.js:

http://railscasts.com/episodes/197-nested-model-form-part-2

Comment (bgeuken) [inline src/api/app/helpers/webui/webui_helper.rb]: btw this could also be set as default id in the parameter

Comment (Ana06) [inline src/api/app/helpers/webui/webui_helper.rb]: @evanrolfe 
> maybe use a timestamp instead? thats how railscasts do it

although railscast is using seconds precision, doubt it is enough here, but with milliseconds it should work

Comment (mdeniz) [inline src/api/app/helpers/webui/webui_helper.rb]: So, I will do something like: 
```ruby
def toggle_sliced_text(text, slice_length = 50, id = "toggle_sliced_text_#{Time.now.to_f.to_s.gsub('.','')}")
```

**Comment by mdeniz** [comment]
@mschnitzer , @evanrolfe , @bgeuken, @DavidKang  and @Ana06 please re-review it 😄

File: src/api/app/helpers/webui/webui_helper.rb [hunk 2]
```diff
@@ -411,4 +411,17 @@ def word_break(string, length)
     # adds a <wbr> tag after an amount of given characters
     safe_join(string.scan(/.{1,#{length}}/), "<wbr>".html_safe)
   end
+
+  def toggle_sliced_text(text, slice_length = 50, id = "toggle_sliced_text_#{Time.now.to_f.to_s.delete('.')}")
+    return text if text.length < slice_length
+    short = content_tag(:span, 'data-toggle-id' => id) do
+      content_tag(:span, text.slice(0, slice_length) + ' ') +
+      link_to('[+]', '#', onclick: "$(\"[data-toggle-id='#{id}']\").toggle();".html_safe)
```
Comment (ChrisBr) [inline src/api/app/helpers/webui/webui_helper.rb]: I really don't like the variable here as if id is some js we have a XSS injection!

```
'$(\"[data-toggle-id='.html_safe + id + '").toggle();'.html_safe
```
would be the safest option!

File: src/api/app/helpers/webui/webui_helper.rb [hunk 3]
```diff
@@ -411,4 +411,17 @@ def word_break(string, length)
...
+    end
+    long = content_tag(:span, 'data-toggle-id' => id, :style => 'display: none;') do
+      content_tag(:span, text + ' ') +
+      link_to('[-]', '#', onclick: "$(\"[data-toggle-id='#{id}']\").toggle();".html_safe)
```
Comment (ChrisBr) [inline src/api/app/helpers/webui/webui_helper.rb]: Same here!

File: src/api/app/helpers/webui/webui_helper.rb (hunk 2 above)
Comment (mdeniz) [inline src/api/app/helpers/webui/webui_helper.rb]: What variable? id?

If so is set by the developer in the view, not the user, isn't it?

**Comment by evanrolfe** [comment]
LGTM :+1:

File: src/api/app/helpers/webui/webui_helper.rb (hunk 2 above)
Comment (ChrisBr) [inline src/api/app/helpers/webui/webui_helper.rb]: Yes, but if some developer uses your function wrong you have an XSS, like

``
def toggle_sliced_text(text, 50, "toggle_#{user.name}_description") 
``

you probably have to sth like:

```
js = '$(\"[data-toggle-id='.html_safe
js << id
js << '").toggle();'.html_safe
```
With my previous comment you would need to do the html_safe on the whole string again!

Comment (ChrisBr) [inline src/api/app/helpers/webui/webui_helper.rb]: If you move js outside of the content block you can even reuse it!

Comment (mdeniz) [inline src/api/app/helpers/webui/webui_helper.rb]: Done! Please review

**Pull request merged**